java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -37,62 +38,97 @@
 public class RangeEqualsVisitor {
 
   protected final ValueVector right;
-  protected final int leftStart;
-  protected final int rightStart;
-  protected final int length;
+  protected int leftStart;
+  protected int rightStart;
+  protected int length;
+
+  protected boolean typeCheckNeeded = true;
+
+  public void setTypeCheckNeeded(boolean typeCheckNeeded) {
+    this.typeCheckNeeded = typeCheckNeeded;
+  }
 
   /**
    * Constructs a new instance.
    */
-  public RangeEqualsVisitor(ValueVector right, int leftStart, int rightStart, int length) {
+  public RangeEqualsVisitor(ValueVector right, int rightStart,  int leftStart, int length, boolean typeCheckNeeded) {
     this.leftStart = leftStart;
     this.rightStart = rightStart;
     this.right = right;
     this.length = length;
+    this.typeCheckNeeded = true;
```

Comment:
this.typeCheckNeeded = typeCheckNeeded ?

Reply:
Right, fixed.